spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/ModuleAggregationUtils.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.aggregate;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.cloud.stream.annotation.EnableModule;
+import org.springframework.cloud.stream.annotation.Processor;
+import org.springframework.cloud.stream.annotation.Sink;
+import org.springframework.cloud.stream.annotation.Source;
+import org.springframework.cloud.stream.binding.BindableContextWrapper;
+import org.springframework.cloud.stream.binding.BindableProxyFactory;
+import org.springframework.cloud.stream.binding.BindingUtils;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.channel.DirectChannel;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class ModuleAggregationUtils {
+
+	private static final Log log = LogFactory.getLog(ModuleAggregationUtils.class);
+
+	/**
+	 * Supports the aggregation of {@link Source}, {@link Sink} and {@link Processor} modules by instantiating and
```

Comment:
In the future, it should work with more general inputs/outputs, but right now it works with only those interfaces. (i.e. it assumes channels named 'input'/'output' - maybe that's what we should say). It would be fair to claim support for more than that,